apps/web/vibes/soul/sections/dynamic-form-section/index.tsx

```diff
@@ -0,0 +1,39 @@
+import { DynamicForm, DynamicFormAction } from '@/vibes/soul/primitives/dynamic-form';
+import { Field, FieldGroup } from '@/vibes/soul/primitives/dynamic-form/schema';
+import { SectionLayout } from '@/vibes/soul/sections/section-layout';
+
+interface Props<F extends Field> {
+  title?: string;
+  subtitle?: string;
+  action: DynamicFormAction<F>;
+  fields: Array<F | FieldGroup<F>>;
+  submitLabel?: string;
+  className?: string;
+}
+
+export function DynamicFormSection<F extends Field>({
+  className,
+  title,
+  subtitle,
+  fields,
+  submitLabel,
+  action,
+}: Props<F>) {
+  return (
+    <SectionLayout className={className} containerSize="lg">
+      {Boolean(title) && (
+        <header className="mx-auto w-full max-w-4xl pb-8 @2xl:pb-12 @4xl:pb-16">
+          <h1 className="mb-5 font-heading text-4xl font-medium leading-none @xl:text-5xl">
+            {title}
+          </h1>
+          {Boolean(subtitle) && subtitle !== '' && (
+            <p className="mb-10 text-base font-light leading-none @xl:text-lg">{subtitle}</p>
+          )}
+        </header>
+      )}
+      <div className="mx-auto w-full max-w-4xl">
```

Comment:
Do we need this div? Max width is being set by the section layout.

Reply:
@apledger Seems like it! Without it, the alignment is off:
![image](https://github.com/user-attachments/assets/b8b582d5-d13b-40b2-9f6e-b3afb09dd31a)

Reply:
It's because we're using `mx-auto` to center the contents. Since `SectionLayout` takes `className` we could use Tailwind there to center the contents by using flexbox or any other centering method. We could also leave this div here but then we can probably drop the `max-w-4xl` class

Reply:
@migueloller Hmm. Whenever I make those changes, it works well in VIBES, but the child `div` in the SectionLayout component has padding by default (and can't be overridden by props), so it still messes up the layout:

![image](https://github.com/user-attachments/assets/98244460-3a24-433d-93e3-d5fb543efe61)

Reply:
I guess in Catalyst we can just not use the `DynamicFormSection` component and stick to the `DynamicForm` by itself 👍